Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/UByte.java

```diff
@@ -0,0 +1,180 @@
+package dafny;
+
+import java.math.BigInteger;
+
+// Dafny bytes are default unsigned, whereas they are signed in Java, and there is no unsigned equivalent
+public class UByte {
+    private int inner;
+    public final static int MAXVALUE = 0xff;
+    public UByte(byte by){
+        //simply casting to an int will preserve the sign
+        inner = Byte.toUnsignedInt(by);
+    }
+
+    public UByte(int i){
+        assert 0 <= i && i <= MAXVALUE : "Precondition Failure";
+        inner = i;
+    }
+
+    public UByte(short s){
+        assert 0 <= s && s <= MAXVALUE : "Precondition Failure";
+        inner = s;
+    }
+
+    public UByte(long l){
+        assert 0 <= l && l <= MAXVALUE : "Precondition Failure";
+        inner = (int) l;
+    }
+
+    public UByte(UByte other){
+        inner = other.inner;
+    }
+
+    public UByte(dafny.UInt other){
+        this(other.value());
+    }
+
+    public UByte(dafny.ULong other){
+        this(other.value());
+    }
+
+    public UByte(dafny.UShort other){
+        this(other.value());
+    }
+
+    public UByte(BigInteger i){
+        assert 0 <= i.compareTo(BigInteger.ZERO) && i.compareTo(BigInteger.valueOf(MAXVALUE)) <= 0 : "Precondition Failure";
+        inner = i.intValue();
+    }
+
+    public static int compare(UByte x, UByte y){
+        return Integer.compareUnsigned(x.inner, y.inner);
+    }
+
+    public int compareTo(UByte other){
+        return Integer.compareUnsigned(inner, other.inner);
+    }
+
+    public double doubleValue(){
+        return (double) inner;
```

Comment:
Why is this included? `double` and `float` are never used, right?

Reply:
They were added just in case they were ever needed.